Missing "enabled" property for addon causes panic

**Is this a request for help?**:

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
Issue
---

**What version of acs-engine?**:
Version: v0.18.8
GitCommit: 71d88da
GitTreeState: clean

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**

Kubernetes 1.10

**What happened**:
An empty Azure resource group gets created

acs-engine Panics
```
acs-engine deploy --resource-group $k8s_resource_group --subscription-id $subscription_id --dns-prefix $k8s_resource_group --location eastus --auto-suffix --api-model GPU_Karl.json
WARN[0003] apimodel: missing masterProfile.dnsPrefix will use "tiny-kmutch-gpu"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6af707]

goroutine 1 [running]:
github.com/Azure/acs-engine/pkg/api/vlabs.(*Properties).validateAddons(0xc42036e100, 0x0, 0x0)
        /go/src/github.com/Azure/acs-engine/pkg/api/vlabs/validate.go:1000 +0x757
github.com/Azure/acs-engine/pkg/api/vlabs.(*Properties).Validate(0xc42036e100, 0x900, 0x103f, 0xc4208e3a18)
        /go/src/github.com/Azure/acs-engine/pkg/api/vlabs/validate.go:472 +0x1bc
github.com/Azure/acs-engine/pkg/api.(*Apiloader).LoadContainerService(0xc4208e3c18, 0xc4200ea600, 0x9a5, 0x103f, 0xc42038d948, 0x5, 0x1, 0x0, 0x11fc260, 0xc42044e160, ...)
        /go/src/github.com/Azure/acs-engine/pkg/api/apiloader.go:171 +0x669
github.com/Azure/acs-engine/pkg/api.(*Apiloader).DeserializeContainerService(0xc4208e3c18, 0xc4200ea600, 0x9a5, 0x103f, 0xc4200e0001, 0x0, 0x103f, 0x0, 0x0, 0x138875e, ...)
        /go/src/github.com/Azure/acs-engine/pkg/api/apiloader.go:45 +0x150
github.com/Azure/acs-engine/cmd.validateApimodel(0xc4208e3c18, 0xc4204105a0, 0xc420227f80, 0x5, 0x0, 0xc4204105a0, 0xc420227f80, 0x5, 0x0)
        /go/src/github.com/Azure/acs-engine/cmd/deploy.go:302 +0xcc
github.com/Azure/acs-engine/cmd.(*deployCmd).load(0xc42037fa20, 0xc4203c6480, 0xc4201b8840, 0x0, 0xb, 0x0, 0x0)
        /go/src/github.com/Azure/acs-engine/cmd/deploy.go:166 +0x2a1
github.com/Azure/acs-engine/cmd.newDeployCmd.func1(0xc4203c6480, 0xc4201b8840, 0x0, 0xb, 0x0, 0x0)
        /go/src/github.com/Azure/acs-engine/cmd/deploy.go:77 +0xca
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).execute(0xc4203c6480, 0xc4201b8790, 0xb, 0xb, 0xc4203c6480, 0xc4201b8790)
        /go/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:647 +0x3e4
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420377d40, 0xc4203c6b40, 0xc4203c6900, 0xc4203c66c0)
        /go/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:726 +0x2d4
github.com/Azure/acs-engine/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420377d40, 0xc42000c018, 0x10e6123)
        /go/src/github.com/Azure/acs-engine/vendor/github.com/spf13/cobra/command.go:685 +0x2b
main.main()
        /go/src/github.com/Azure/acs-engine/main.go:12 +0x74
```
**What you expected to happen**:

k8s style autoscaling is deployed

**How to reproduce it** (as minimally and precisely as possible):

Use the following json file.
```
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.10",
      "kubernetesConfig": {
            "addons": [
                {
                    "name": "cluster-autoscaler",
                    "containers": [
                    {
                        "name": "cluster-autoscaler",
                        "cpuRequests": "100m",
                        "memoryRequests": "300Mi",
                        "cpuLimits": "100m",
                        "memoryLimits": "300Mi"
                    }
                    ],
                    "config": {
                    "maxNodes": "2",
                    "minNodes": "1"
                    }
                }
            ]
        }
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "",
      "vmSize": "Standard_D2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "agentpool1",
        "count": 2,
        "vmSize": "Standard_ND6s",
        "OSDiskSizeGB": 200,
        "availabilityProfile": "AvailabilitySet"
      }
    ],
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [
          {
            "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
     "clientId": "*********",
     "secret": "**********"
    },
    "certificateProfile": {}
  }
}
```
Run the command:
```
acs-engine deploy --resource-group $k8s_resource_group --subscription-id $subscription_id --dns-prefix $k8s_resource_group --location eastus --auto-suffix --api-model GPU_Karl.json
```

**Anything else we need to know**:

Using other addons works, along with using no addons.

--- a/docs/clusterdefinition.md
+++ b/docs/clusterdefinition.md
@@ -113,6 +113,7 @@ More usefully, let's add some custom configuration to the above addons:
     "addons": [
         {
             "name": "tiller",
+            "enabled": true,
             "containers": [
                 {
                   "name": "tiller",
@@ -126,6 +127,7 @@ More usefully, let's add some custom configuration to the above addons:
         },
         {
             "name": "kubernetes-dashboard",
+            "enabled": true,
             "containers": [
                 {
                   "name": "kubernetes-dashboard",
@@ -138,6 +140,7 @@ More usefully, let's add some custom configuration to the above addons:
         },
         {
             "name": "cluster-autoscaler",
+            "enabled": true,
             "containers": [
               {
                 "name": "cluster-autoscaler",

--- a/pkg/api/vlabs/validate.go
+++ b/pkg/api/vlabs/validate.go
@@ -409,7 +409,7 @@ func (a *Properties) validateAddons() error {
 		for _, addon := range a.OrchestratorProfile.KubernetesConfig.Addons {
 			switch addon.Name {
 			case "cluster-autoscaler":
-				if *addon.Enabled && isAvailabilitySets {
+				if helpers.IsTrueBoolPointer(addon.Enabled) && isAvailabilitySets {
 					return fmt.Errorf("Cluster Autoscaler add-on can only be used with VirtualMachineScaleSets. Please specify \"availabilityProfile\": \"%s\"", VirtualMachineScaleSets)
 				}
 			case "nvidia-device-plugin":

